rules/ast/ast.go

```diff
@@ -241,12 +241,23 @@ type (
 	// least one of the two operand Nodes must be a VectorNode. The other may be
 	// a VectorNode or ScalarNode. Both criteria are checked at runtime.
 	VectorArithExpr struct {
-		opType BinOpType
-		lhs    Node
-		rhs    Node
+		opType        BinOpType
+		lhs           Node
+		rhs           Node
+		matchType     VectorMatchType
+		matchBy       clientmodel.LabelNames
+		includeLabels clientmodel.LabelNames
 	}
 )
 
+type VectorMatchType int
+
+const (
+	MatchOTO VectorMatchType = iota
```

Comment:
For the sake of clarity, could we name these either:

`MatchOneToOne` / `MatchManyToOne` / ...

or:

`Match1To1` / `Match1ToN` / ...

(I think I prefer the former)